### PR TITLE
Update EOL versions

### PR DESCRIPTION
**Pull request by richlander**

### CODE REVIEW

**Review by jander-msft — DISMISSED** [review]

**Comment by richlander** [comment]
Updated queue names, same as https://github.com/dotnet/dotnet-monitor/pull/6281

File: eng/helix/Helix.proj [hunk 1]
```diff
@@ -58,38 +58,38 @@
...
   <ItemGroup Condition="$([MSBuild]::IsOSPlatform(Linux)) and !$(PackageRid.Contains(musl))">
     <!-- linux-arm64 -->
-    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-arm64v8"
+    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-12-helix-arm64v8"
                                Condition="'$(HelixArchitecture)' == 'arm64'">
-      <TestRunName>Debian 11 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+      <TestRunName>Debian 12 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
     </HelixAvailableTargetQueue>
-    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8"
+    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-helix-arm64v8"
```
Comment (jander-msft) [inline eng/helix/Helix.proj:65]: @richlander, there is no `ubuntu-20.04-helix-arm64v8` tag for `mcr.microsoft.com/dotnet-buildtools/prereqs`. See https://mcr.microsoft.com/v2/dotnet-buildtools/prereqs/tags/list. In fact, there is no helix image for Ubuntu 20.04 on arm64. This is a little surprising given that .NET 8 is supported on 20.04 and above: https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#linux.

Comment (richlander) [reply eng/helix/Helix.proj:65]:
```suggestion
    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-arm64v8"
```

Let's just use 22.04 for Arm64 then. We could create 20.04 image, but it probably doesn't matter. We're covering the full combination of 20.04 and Arm64 and x64 (and get 22.04 for free). Seems fine to me.

File: eng/helix/Helix.proj [hunk 2]
```diff
@@ -58,38 +58,38 @@
...
+    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-12-helix-arm64v8"
                                Condition="'$(HelixArchitecture)' == 'arm64'">
-      <TestRunName>Debian 11 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+      <TestRunName>Debian 12 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
     </HelixAvailableTargetQueue>
-    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8"
+    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-arm64v8"
                                Condition="'$(HelixArchitecture)' == 'arm64'">
-      <TestRunName>Ubuntu 18.04 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
+      <TestRunName>Ubuntu 20.04 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```
Comment (jander-msft) [inline eng/helix/Helix.proj:67]:
```suggestion
      <TestRunName>Ubuntu 22.04 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```

File: eng/helix/Helix.proj [hunk 3]
```diff
@@ -58,38 +58,38 @@
...
     </HelixAvailableTargetQueue>
   </ItemGroup>
 
   <!-- Linux musl libc queues -->
   <ItemGroup Condition="$([MSBuild]::IsOSPlatform(Linux)) and $(PackageRid.Contains(musl))">
     <!-- linux-musl-arm64 -->
-    <HelixAvailableTargetQueue Include="ubuntu.1804.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.15-helix-arm64v8"
+    <HelixAvailableTargetQueue Include="ubuntu.2204.armarch$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.17-helix-arm64v8"
                                Condition="'$(HelixArchitecture)' == 'arm64'">
       <TestRunName>Alpine 3.15 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```
Comment (jander-msft) [inline eng/helix/Helix.proj:89]:
```suggestion
      <TestRunName>Alpine 3.17 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```

File: eng/helix/Helix.proj [hunk 4]
```diff
@@ -58,38 +58,38 @@
...
     </HelixAvailableTargetQueue>
     <!-- linux-musl-x64 -->
-    <HelixAvailableTargetQueue Include="ubuntu.1804.amd64$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.15-helix-amd64"
+    <HelixAvailableTargetQueue Include="ubuntu.2204.amd64$(QueueSuffix)@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.17-helix-amd64"
                                Condition="'$(HelixArchitecture)' == 'x64'">
       <TestRunName>Alpine 3.15 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```
Comment (jander-msft) [inline eng/helix/Helix.proj:94]:
```suggestion
      <TestRunName>Alpine 3.17 $(HelixArchitecture) $(HelixConfiguration)</TestRunName>
```

**Review by jander-msft — APPROVED** [review]

**Pull request merged**